### DISCUSSION TITLE
util for reset context to their initial values

### DISCUSSION THREAD

**Discussion by cringelord1940** [statelyai/xstate#5112] [Ideas]
I find this to be useful for a software flow to rerun

**Comment by davidkpiano** [comment]
You can do this by storing the static context outside of the machine:

```ts
const initialContext = { count: 0 }

const machine = setup({ ... }).createMachine({
  context: initialContext
})
```

For input, you can store the input on context:

```ts
const machine = setup({ ... }).createMachine({
  context: ({ input }) => ({
    input,
    // the rest of your context
  })
})
```

And use that `context.input` to reset your context.

A built-in "reset" action would mean that we'd have to implicitly store that initial context inside the state, and it's probably better for that kind of state to be explicitly stored by the developer.
